src/browser/starter.js

```diff
@@ -1121,6 +1121,46 @@ V86.prototype.serial_send_bytes = function(serial, data)
     }
 };
 
+/**
+ * Set the modem status of a serial port.
+ */
+V86.prototype.serial_set_modem_status = function(serial, status)
+{
+    this.bus.send("serial" + serial + "-modem-status-input", status);
+};
+
+/**
+ * Set the carrier detect status of a serial port.
+ */
+V86.prototype.serial_set_carrier_detect = function(serial, status)
+{
+    this.bus.send("serial" + serial + "-carrier-detect-input", status);
+};
+
+/**
+ * Set the ring indicator status of a serial port.
+ */
+V86.prototype.serial_set_ring_indicator = function(serial, status)
+{
+    this.bus.send("serial" + serial + "-ring-indicator-input", status);
+};
+
+/**
+ * Set the data set ready status of a serial port.
+ */
+V86.prototype.serial_set_data_set_ready = function(serial, status)
+{
+    this.bus.send("serial" + serial + "-data-set-ready-input", status);
+};
+
+/**
+ * Set the clear to send status of a serial port.
+ */
+V86.prototype.serial_set_clear_to_send = function(serial, status)
+{
+    this.bus.send("serial" + serial + "-clear-to-send-input", status);
+};
```

Comment:
This wouldn't prevent merging, but do you have a test for this? (either some program that actually calls these or a unit test that calls these functions and checks some output from the guest OS)

Reply:
I spent tome time on this, but haven't been able to find an easy way to test this. It's easy enough to add an api-test along with serial.js, but the problem is that I am having trouble finding current guest OS that have support for checking these flags built-in (i.e. minicom, etc.) I could either write a C program to check them, or create a new/modified image that has a program that cares about these flags, but I'm having trouble finding one that already has something that would work for this (at least with the LInux and DOS images, I'm not as familiar with some of the others to see.) If you have any suggestions for how to proceed I'd be happy to do so, otherwise I'm afraid I don't have one. (I do have external programs using my own images that I've been testing with, but none of the current ones from v86.)

Reply:
No problem!